RijsMuseum/app/src/main/java/com/example/rijsmuseum/fragments/DetailsFragment.kt

```diff
@@ -32,12 +33,14 @@ class DetailsFragment : Fragment(R.layout.fragment_collection_details) {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         titleText = view.findViewById(R.id.fragmentTitle)
-
-        detailsViewModel.getCollectionsDetailsRequest(object : CollectionsDetailsGetter.DataReadyCallback {
-            override fun onDataReady(data: CollectionsDetails.ArtObject) {
-                titleText.text = data.title
-            }
-
-        })
+        requireArguments().getString("objectNumber")?.let {
+            Log.d("Bundle", "Object number is: $it")
+            detailsViewModel.getCollectionsDetailsRequest(object : CollectionsDetailsGetter.DataReadyCallback {
+                override fun onDataReady(data: CollectionsDetails.ArtObject) {
+                    titleText.text = data.title
```

Comment:
move the callback into the view model. Then create a LiveData and get the object through the LiveData